File: RGBEngine/src/main/java/engine/util/Resources.java
package engine.util;

import java.io.IOException;
import java.net.URL;

public abstract class Resources {

    /**
     * Loads the contents of the file at the given URL into an array of bytes.
     * Throws an IOException if we failed to load the file.
     * Throws a NullPointerException if the given url is null.
     *
     * @param url The URL to load from
     * @return The array of bytes in the file
     */
    public static byte[] loadBytes(URL url) {
        try {
            return url.openStream().readAllBytes();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    /**
     * Loads the contents of the file at the given URL into a String.
     * Throws an IOException if we failed to load the file.
     * Throws a NullPointerException if the given url is null.
     *
     * @param url The URL to load from
     * @return The text in the file
     */
    public static String loadString(URL url) {
        return new String(loadBytes(url));
    }

    /**
     * Loads the contents of the file at the given URL into a String.
     * Returns null if the given URL is null.
     * Throws an IOException if we failed to load the file.
     *
     * @param url The URL to load from
     * @return The text in the file, or null if the given URL is null
     */
    public static String loadStringNullable(URL url) {
        return url == null ? null : loadString(url);
    }
}

File: Cookie/src/main/kotlin/engine/util/Resources.kt
package engine.util

import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths

object Resources {

    fun loadFileAsBytes(path: String): ByteArray {
        try {
            return Files.readAllBytes(Paths.get(path))
        } catch (ex: IOException) {
            System.out.println(Paths.get(path).toAbsolutePath())
            throw RuntimeException(ex)
        }

    }

    fun loadFileAsString(path: String): String {
        return String(loadFileAsBytes(path))
    }
}
